### ISSUE TITLE
Wrong unit icon if the secondary player is not specified in the SCN file

### ISSUE THREAD

**Issue by salexgit** [nicupavel/openpanzer#184]
Hi Nicu

While converting and testing one of the German campaigns, I noticed that the game engine for some reason displays wrong unit icons for the secondary player. Here is the background information...

As you can see in the attached screenshot taken from the SCN file editor, there is a group of Romanian units (which is the secondary player in this German scenario). The selected unit is a cannon with three infantry units.

![pg2suite](https://user-images.githubusercontent.com/22852936/49679032-20113280-fa91-11e8-8566-2a2d5f92e64b.png)

However, when I opened the same scenario in the OpenPanzer game, I noticed that all the units turn into some Flaks, as you can see in the screenshot below.

![openpanzer](https://user-images.githubusercontent.com/22852936/49679051-3dde9780-fa91-11e8-98ca-20dc50e065a8.png)

Initially I thought that there is a problem with the conversion script. However, XML file has correct country and the unit ID information. My next guess was that the problem is due to a wrong equipment file or mismatching equipment files, but they are perfectly fine. The Romanian cannon unit ID is 1575 and Romanian infantry unit ID is 308; and all of them exist in the Romanian "equipment-country-18.json" file. So, the only explanation I have is that game engine does not read the correct equipment file for the secondary player and that is why unit icon and type is completely wrong. Can you please check it?

To ease the debugging efforts, I am attaching SCN, MAP, and XML files. If you need some help in troubleshooting, I can help, but I do not have an access to the game engine.

[lah_it_09.zip](https://github.com/nicupavel/openpanzer/files/2659192/lah_it_09.zip)

**Comment by nicupavel** [comment]
looks like half of them are showing correctly and half incorrectly ? or the ones that show correctly are german ?

**Comment by salexgit** [comment]
H Nicu

The correct ones are German units, i.e. the ones belonging to the main player. It is only the secondary player units which are wrong. If I am not mistaken I saw this problem also in other scenarios, so I will provide more information once I recall where exactly it was.

**Comment by salexgit** [comment]
Hi again

I did further checking and I am pretty sure that the root cause is that the game engine uses wrong equipment file ID when units are not from the same country as the main play country is. There was a bug in one scenario mistakenly indicating "Soviets" for the Wacht Am Rhein scenario. As you can see from the attached file, the same problem emerged.

![wamr1](https://user-images.githubusercontent.com/22852936/49692015-21f1f900-fb58-11e8-9500-f6bbc6b34495.png)

Once I fixed the main player ID to "USA", all the units icons became correct.

![wamr2](https://user-images.githubusercontent.com/22852936/49692025-40f08b00-fb58-11e8-87a7-c35d8c47e39c.png)

P.S. I have been using v2.9.7 for testing but I assume that the same problem exists in v3.0.0

**Comment by nicupavel** [comment]
I opened the lah_it_09 scn in SuitePG2 but it does not have a supporting country setting either for Axis or Allied.

Supporting countries should also show up on equipment window if you click on the country flag (it should change to next supporting country)

**Comment by salexgit** [comment]
Hi Nicu

Thanks a lot for checking! I will assign Axis to the supporting country and will check whether it fixes the issue.

P.S.: It seems that PG2 has been less sensitive to all those minor discrepancies because I always need to do some SCN file fixes.

**Comment by nicupavel** [comment]
You need to assign Romania as supporting country. OpenPanzer loads equipment per country to use less memory whereas PG2 loads all equipment anyway.

**Comment by salexgit** [comment]
Hi

Ok, now I see why wrong/garbage icons are displayed if the corresponding country is now specified explicitly in the SCN file. Thanks!

**Comment by salexgit** [comment]
Hi

Ok, once I specified explicitly Romania as the secondary player country, all the units are displayed correctly.

I understand that it is nice/important to have consistent SCN files, but mistakes like that are a bit complex to detect; and it is even more challenging to debug them without knowing how the game engine actually works. I noticed that some SCN files simply do not specify correctly the secondary country at all.
Can I ask you to add some "assertion" message into the code? If the game engine encounters a unit, country equipment file of which is not loaded, then at least some error message can be shown. This message can be added to a "debug" mode if you have it.

Regards, Alex
